### ISSUE TITLE
select is not updating when using with mut on ember 3.20.4

### ISSUE THREAD

**Issue by Kilowhisky** [cibernox/ember-power-select#1398]
Since updating to ember `3.20.4` the select list no longer updates when changes to the model occur if used against an ember data hasMany.  

![ezgif-7-4a53d80dc774](https://user-images.githubusercontent.com/7075666/93125119-87eda780-f67f-11ea-9bd5-20fbf88ae1c3.gif)

Here is a demo repo showing the issue: https://github.com/Kilowhisky/ember-array-proxy-bug 

This may be related to 

https://github.com/emberjs/ember.js/issues/19101
https://github.com/emberjs/ember.js/pull/19111
https://github.com/emberjs/data/issues/7281

**Comment by wuarmin** [comment]
@Kilowhisky @cibernox 
I observed a similar issue with Ember 3.20.5. I spend a lot of time for investigation.

![Peek 2020-10-05 13-06](https://user-images.githubusercontent.com/9987680/95072579-b523f780-070b-11eb-88e7-31f8e21cc208.gif)

**Comment by cibernox** [comment]
I missed this when it was originally reported. If it does work with 3.21 I can only assume it's a bug in Ember or Ember-data itself? Also, if worked before 3.20.4 and works again starting in 3.21.X, I'd rather not introduce any workaround for an ember bug that was present in only two minor patch versions.

Sounds fair?

**Comment by wuarmin** [comment]
@cibernox of course, that sounds fair. I hope there's a patch for 3.20, because it's a LTS version.

Best regards

**Comment by benreckas** [comment]
Was experiencing this issue with Ember v `3.21.2` and Ember Data v `3.21.0`. Updating to `3.22` for both fixed it in my case.
